model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -184,55 +187,72 @@ message ProcessBundleDescriptor {
   org.apache.beam.model.pipeline.v1.ApiServiceDescriptor state_api_service_descriptor = 7;
 }
 
-// Represents a partition of the bundle into two bundles: a "primary" and
-// a "residual", with the following properties:
-// - The work in primary and residual doesn't overlap, and combined, adds up
-//   to the work in the current bundle if the split hadn't happened.
-// - The current bundle, if it keeps executing, will have done none of the
-//   work under residual roots.
-// - The current bundle, if no further splits happen, will have done exactly
-//   the work under primary_roots.
-// For more rigorous definitions see https://s.apache.org/beam-breaking-fusion
-message BundleSplit {
-  // One of the root applications specifying the scope of work for a bundle.
-  message Application {
-    // (Required) The primitive transform to which to pass the element
-    string ptransform_id = 1;
+// One of the applications specifying the scope of work for a bundle.
+message BundleApplication {
+  // (Required) The primitive transform to which to pass the element
+  string ptransform_id = 1;
```

Comment:
In this layer of the architecture do we talk about the difference between 'PTransform' and 'applied PTransform'? If I understand correctly, PTransform is an SDK concept which can be applied many times in a pipeline; and a bundle application refers to a specific application of a PTransform.

Reply:
Applied PTransforms are an internal representation detail of how an SDK tracks "duplication" of the same PTransform within a graph.

From the Beam Portability APIs, the pipeline proto only represent applied PTransforms as their are no unapplied PTransforms.